api/core/database.py

```diff
@@ -28,6 +28,27 @@ def save(self, email, raw_password):
         db.session.commit()
         return self
 
+    @classmethod
+    def verify_user(cls, email, password):
+        email = email.strip()
+        user = cls.query.filter_by(mail=email).first()
+        cls._verificate_user(email, password, user)
+        return True
+
+    @staticmethod
+    def _verificate_user(email, password, user_object):
+        errors = {}
+
+        if not User.query.filter_by(mail=email).first():
+            errors['mail'] = 'user does not exist'
+
+        else:
+            if not checkpw(password.encode(), user_object.password):
+                errors['password'] = 'incorrect password'
+
+        if errors:
+            raise VerificationError(errors)
```

Comment:
Многовато получилось кода, давай упрощать
```suggestion
    @classmethod
    def is_exists(cls, email, password):
        email = email.strip()
        user = cls.query.filter_by(mail=email).first()
        if not user:
            return False
            
        return checkpw(password.encode(), user.password)
```
И убери `VerificationError`. Я понимаю, что ты хотел сделать по аналогии с `ValidationError`, но это немного разные случаи. `ValidationError` - это комплексная провека и валидация данных (в твоем случае не слишком сложная и проверок не так много, но идейно - это именно комплексная проверка)

`VerificationError` - это проверка одного конкретного случая "есть пользователь в базе или нет". Для одной проверки нет смысла делать новое исключение с полем `errors`

Reply:
Там две проверки. Есть ли в базу пользователь, и если есть, верен ли его пароль.

Reply:
Я бы предложил не заморачиваться с 2мя ошибками. Это все-таки API, а не форма веб-интерфейса. Это конечно спорный вопрос, но считают не безопасным писать, что неверен именно пароль, это дает возможность перебора пользователей по email.

Reply:
> но считают не безопасным писать, что неверен именно пароль

Тоже об этом подумал. Но на практике 50 на 50. Можно будет капчу воткнуть при 3х попытках. Или все таки переделаем?